rcp/org.testeditor.rcp4.views.projectexplorer.tests/src/org/testeditor/rcp4/views/projectexplorer/TEContentProviderTest.xtend

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.views.projectexplorer
+
+import org.eclipse.core.resources.IProject
+import org.eclipse.jdt.core.JavaCore
+import org.junit.Test
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.testeditor.dsl.common.testing.AbstractTest
+
+import static org.mockito.Mockito.*
+import org.eclipse.jdt.core.IJavaProject
+import org.eclipse.jdt.core.IClasspathEntry
+import org.eclipse.core.resources.IWorkspaceRoot
+import org.eclipse.core.resources.IFolder
+import org.eclipse.core.resources.IResource
+
+class TEContentProviderTest extends AbstractTest {
```

Comment:
The `// given` steps are huge for the simple things we want to test here. Can we simplify this somehow?